### DISCUSSION TITLE
/tmp/cache filling up

### DISCUSSION THREAD

**Discussion by YBonline** [blakeblackshear/frigate#5707] [Ask A Question] [answered]
So as I continue my journey with Frigate and adding additional cameras to it, I've noticed the /tmp/cache directory is filling up, leading to it dropping recordings. I have 20 cameras now on it, and it was randomly dropping some recordings. I had some memory issues (unrelated to frigate), but have resolved them, and the problem continued, so now I realized /tmp/cache was filling up. So I increased it from 4GB to 8GB and now to 28GB, and it still seems to fill up and drop a similar number of recordings as when it was set to 4GB. How is the size of /tmp/cache supposed to be calculated, and is there something else I'm missing? I noticed some cameras have multiple files in the cache: I'm on 0.12 beta9

root@663529866bcc:/tmp/cache# ls -alh
total 28G
drwxr-xr-x 2 root root  700 Mar 12 18:23 .
drwxrwxrwt 1 root root 4.0K Mar 12 08:46 ..
prwxr-xr-x 1 root root    0 Mar 12 08:47 birdseye
-rw-r--r-- 1 root root 2.6M Mar 12 18:23 gp525_doorbell-20230312182253.mp4
-rw-r--r-- 1 root root 257K Mar 12 18:23 gp525_doorbell-20230312182303.mp4
-rw-r--r-- 1 root root 3.4G Mar 12 18:23 gp636_doorbell-20230312084752.mp4
-rw-r--r-- 1 root root 1.9G Mar 12 18:23 gp_clubhouse_back_door-20230312163825.mp4
-rw-r--r-- 1 root root 2.1G Mar 12 18:23 gp_clubhouse_banquet-20230312163825.mp4
-rw-r--r-- 1 root root 4.7G Mar 12 18:23 gp_clubhouse_firepit-20230312163826.mp4
-rw-r--r-- 1 root root 6.1G Mar 12 18:23 gp_clubhouse_front_door-20230312131316.mp4
-rw-r--r-- 1 root root 3.7G Mar 12 18:23 gp_clubhouse_gym_1-20230312163829.mp4
-rw-r--r-- 1 root root 5.9M Mar 12 18:22 gp_clubhouse_gym_2-20230312182244.mp4
-rw-r--r-- 1 root root 5.3M Mar 12 18:23 gp_clubhouse_gym_2-20230312182254.mp4
-rw-r--r-- 1 root root 3.4G Mar 12 18:23 gp_clubhouse_parking-20230312110615.mp4
-rw-r--r-- 1 root root 4.5M Mar 12 18:22 gp_clubhouse_pool-20230312182248.mp4
-rw-r--r-- 1 root root 3.1M Mar 12 18:23 gp_clubhouse_pool-20230312182257.mp4
-rw-r--r-- 1 root root 2.2M Mar 12 18:22 gp_clubhouse_pool_porch-20230312182243.mp4
-rw-r--r-- 1 root root 2.2M Mar 12 18:23 gp_clubhouse_pool_porch-20230312182253.mp4
-rw-r--r-- 1 root root   48 Mar 12 18:23 gp_clubhouse_pool_porch-20230312182303.mp4
-rw-r--r-- 1 root root 189K Mar 12 18:22 warehouse_exterior_doorbell-20230312182242.mp4
-rw-r--r-- 1 root root 154K Mar 12 18:23 warehouse_exterior_doorbell-20230312182254.mp4
-rw-r--r-- 1 root root   48 Mar 12 18:23 warehouse_exterior_doorbell-20230312182303.mp4
-rw-r--r-- 1 root root 827M Mar 12 18:23 warehouse_exterior_front-20230312163839.mp4
-rw-r--r-- 1 root root 3.8M Mar 12 18:23 warehouse_exterior_nw-20230312182251.mp4
-rw-r--r-- 1 root root 769K Mar 12 18:23 warehouse_exterior_nw-20230312182301.mp4
-rw-r--r-- 1 root root 2.1G Mar 12 18:23 warehouse_exterior_se-20230312163825.mp4
-rw-r--r-- 1 root root 3.7M Mar 12 18:23 warehouse_exterior_sw-20230312182252.mp4
-rw-r--r-- 1 root root 513K Mar 12 18:23 warehouse_exterior_sw-20230312182302.mp4
-rw-r--r-- 1 root root 7.3M Mar 12 18:22 warehouse_interior_front-20230312182243.mp4
-rw-r--r-- 1 root root 5.6M Mar 12 18:23 warehouse_interior_front-20230312182255.mp4
-rw-r--r-- 1 root root 257K Mar 12 18:22 warehouse_interior_office-20230312182239.mp4
-rw-r--r-- 1 root root 3.5M Mar 12 18:23 warehouse_interior_overview-20230312182251.mp4
-rw-r--r-- 1 root root 513K Mar 12 18:23 warehouse_interior_overview-20230312182301.mp4
-rw-r--r-- 1 root root 6.1M Mar 12 18:22 warehouse_workbench-20230312182245.mp4
-rw-r--r-- 1 root root 5.8M Mar 12 18:23 warehouse_workbench-20230312182254.mp4

**Comment by stevenbrown1862** [comment, marked as answer]
If you are on Beta9 then a few folks, including me have had this issue or similar.  I don't know how to link to it but have a look at look at the the issue thread titled 'beta 9 not saving clips for some cameras'  as there is a fix in there.

**Comment by YBonline** [comment]
Thanks, that does appear to be my issue! The solution doesn't seem to be working for me, likely because the hwaccel options are wrong for my setup, unfortunately I have quite a busy week so I don't have time to investigate further, so hopefully it will get resolved in the next beta before I have time to figure it out!

**Reply by mrplow** [reply]
I had the same problem too, issue #5659
See @blakeblackshear's pull request #5715 
It reverts changes to global_args and output_args detect by removing -threads 1
You can edit frigate/config.py to match the pull request, wait for beta10, or add the override options at the bottom of issue #5677 to your config

@stevenbrown1862 to link to an issue just type in #issue number then check that it worked in the Preview 👍

**Reply by YBonline** [reply]
Interesting. That threads 1 change did cut my memory usage down on my gpu by almost 20% so thats too bad!

**Reply by blakeblackshear** [reply]
You could try setting a thread count to 5 and see if your cache still fills up.

**Reply by YBonline** [reply]
I can confirm my cache does not fill up with threads 5, my GPU memory usage goes from 52% to 69%. I also tried with thread count set to 2, which sets my memory usage at 58% with the cache usage still remaining very small (<100MB). Perhaps this is a good compromise default to allow the memory/resource savings of a lower thread count? I will keep my thread count at 2 for the next couple days and report back what happens

(This is all with a NVIDIA T1000 8GB GPU and 20 cameras, and likely to be at 25 cameras at some point this weekend)
